tls/s2n_handshake_io.c

```diff
@@ -1289,7 +1289,7 @@ static int s2n_handshake_read_io(struct s2n_connection *conn)
          *# SHOULD be ignored by the client if it arrives in the middle of a handshake.
          */
         if (message_type == TLS_HELLO_REQUEST) {
-            POSIX_GUARD(s2n_client_hello_request_recv(conn));
+            POSIX_GUARD_RESULT(s2n_client_hello_request_validate(conn));
```

Comment:
Do we handle this already:

> This message[HelloRequest] MUST NOT be included in the message hashes that are
>    maintained throughout the handshake and used in the Finished messages
>    and the certificate verify message.

Reply:
We do, but choosing where to put the compliance comment might be tricky :) Because we wipe io and continue right after we validate the message, we never get to [s2n_finish_read](https://github.com/aws/s2n-tls/blob/3c0b637b1219413f98237e2690be09d710f0af88/tls/s2n_handshake_io.c#L1305), so we never [update the handshake hashes](https://github.com/aws/s2n-tls/blob/3c0b637b1219413f98237e2690be09d710f0af88/tls/s2n_handshake_io.c#L1144).